### PR TITLE
Board builder

### PR DESCRIPTION
**Pull request by clinton56**
Please make a review.

### CODE REVIEW

File: src/main/java/io/github/expansionteam/battleships/engine/board/Board.java [hunk 1]
```diff
@@ -3,8 +3,8 @@
 import java.util.Iterator;
 import java.util.Map;
 
-public class Board implements Iterable {
-    private Map<Position, Field> map;
+public class Board implements Iterable<Field> {
+    Map<Position, Field> map;
```
Comment (tdudzik) [inline src/main/java/io/github/expansionteam/battleships/engine/board/Board.java]: Map should be private. Btw we can name this fieldsByPosition.

File: src/main/java/io/github/expansionteam/battleships/engine/board/BoardBuilder.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import java.util.Map;
+import java.util.TreeMap;
+
+public class BoardBuilder implements IBoardBuilder {
+
+    private Map<Position, Field> fields = new TreeMap<>();
+    private int size;
+
+    public BoardBuilder(BoardSize boardSize) {
+        size = boardSize.getSize();
```
Comment (tdudzik) [inline src/main/java/io/github/expansionteam/battleships/engine/board/BoardBuilder.java]: maybe boardSize.getValue()?

File: src/main/java/io/github/expansionteam/battleships/engine/board/Board.java (hunk 1 above)
Comment (clinton56) [inline src/main/java/io/github/expansionteam/battleships/engine/board/Board.java]: Yes, I making rebase I overlooked it ;)

File: src/test/java/io/github/expansionteam/battleships/engine/board/BoardBuilderTest.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+import org.testng.annotations.Test;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.testng.Assert.assertEquals;
+
+public class BoardBuilderTest {
+
+    @Test
+
```
Comment (tdudzik) [inline src/test/java/io/github/expansionteam/battleships/engine/board/BoardBuilderTest.java]: Why space? :D This is the best code you've ever written !!!

File: src/test/java/io/github/expansionteam/battleships/engine/board/BoardBuilderTest.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    public void builBoardFromBoardSize() {
+        // given
+        BoardSize boardSizeMock = mock(BoardSize.class);
+        when(boardSizeMock.getSize()).thenReturn(3);
+
+        BoardBuilder boardBuilder = new BoardBuilder(boardSizeMock);
+
+        // when
+        boardBuilder.addFields();
+        Board board = boardBuilder.build();
```
Comment (tdudzik) [inline src/test/java/io/github/expansionteam/battleships/engine/board/BoardBuilderTest.java]: Maybe enter here?

**Pull request merged**